content/learn/book/graphics/rendering-internals/technical-summary.md

```diff
@@ -0,0 +1,849 @@
+[`wgpu::Adapter`]: https://docs.rs/wgpu/latest/wgpu/struct.Adapter.html
+[`wgpu::Device`]: https://docs.rs/wgpu/latest/wgpu/struct.Device.html
+
+### Buffers
+Allocating memory is the most basic of all operations. In `wgpu`, GPU memory
+management is done through buffers.
+
+A `wgpu::Buffer` represents a continuous block of memory allocated on the GPU.
+Buffers can be created by calls to [`wgpu::Device::create_buffer`] (which only
+allocates the buffer) or [`wgpy::DeviceExt::create_buffer_init`] (which both
+allocates and initializes the buffer).
+
+You can reference a slice of a buffer using [`wgpu::Buffer::slice`]. This
+returns a [`wgpu::BufferSlice`] which can be "mapped" to main program memory for
+reading and writing. Writes are flushed by calling [`wgpu::Buffer::unmap`] and
+can be deallocated with [`wgpu::Buffer::destroy`]. Writes can also be queued as part 
+of commands using [`wgpu::Queue::write_buffer`].
+
+When you create a buffer, you generally have to declare the intended use with
+[`wgpu::BufferUses`]. Buffers must have `BufferUsages::MAP_READ` to be read from.
+
+### Textures, Views, and Samplers
+A `wgpu::Texture` represents an image stored in GPU memory. Textures can be
+created by calling `wgpu::Device::create_texture`, which requires information
+about the format, dimensions, and so on. The main way to write to a texture is
+[`wgpu::Buffer::write_texture`], which is similar to
+[`wgpu::Device::write_buffer`] in that it is delayed for dispatch until commands
+are next sent to the GPU. It's also possible to copy a buffer to a texture using
+[`wgpu::CommandEncoder::copy_buffer_to_texture`]. Like buffers, textures can be
+deallocated using [`wgpu::Texture::destroy`].
+
+Textures are usually accessed in shaders through views and samplers.
+
+A [`wgpu::TextureView`] is a "view" on a specific texture. The view describes
+how the raw texture data should be interpreted, and the subset of the texture to
+access. Views are created using [`wgpu::Texture::create_view`]. They will be
+very important when we get into bind groups later on.
+
+A [`wgpu::Sampler`] is a method of sampling color values from a texture at a
+given texture coordinate. They are created using
+[`wgpu::Device::create_sampler`], and will also be important in the context of
+bind groups.
+
+### Pipelines and Shaders
+A pipeline is like a highly-specialized program that runs on the GPU.
+Currently `wgpu` supports two types of pipeline: [`wgpu::RenderPipeline`] for
+rasterizing geometry, and [`wgpu::ComputePipeline`] for general computation.
+
+Pipelines are composed of a sequence of stages, which can be either fixed or
+programmable. Fixed stages are inflexible hardware-accelerated operations, with
+no user control. Programmable stages, on the other hand, can be controlled by
+user-supplied shader modules.
+
+#### Shaders
+Shaders are small highly-specialized programs that run on the GPU within the
+programmable stages of pipelines. They are written in a language `WGSL`.
```

Comment:
"They are written in the [`WGSL`](https://www.w3.org/TR/WGSL/) language"

link to https://www.w3.org/TR/WGSL/ or maybe the book page on it